db/db.go

```diff
@@ -39,32 +39,45 @@ var logger = logrus.StandardLogger().WithField("module", "db")
 var epochsCache = cache.New(time.Hour, time.Minute)
 var saveValidatorsMux = &sync.Mutex{}
 
-func mustInitDB(writer *types.DatabaseConfig, reader *types.DatabaseConfig) (*sqlx.DB, *sqlx.DB) {
-	dbConnWriter, err := sqlx.Open("pgx", fmt.Sprintf("postgres://%s:%s@%s:%s/%s?sslmode=disable", writer.Username, writer.Password, writer.Host, writer.Port, writer.Name))
-	if err != nil {
-		logger.Fatal(err)
-	}
-
+func dbTestConnection(dbConn *sqlx.DB, dataBaseName string) {
 	// The golang sql driver does not properly implement PingContext
 	// therefore we use a timer to catch db connection timeouts
+	endTimeoutDetection := make(chan bool)
 	dbConnectionTimeout := time.NewTimer(15 * time.Second)
+
 	go func() {
-		<-dbConnectionTimeout.C
-		logger.Fatalf("timeout while connecting to the database")
+		for {
+			select {
+			case <-dbConnectionTimeout.C:
+				logger.Fatalf("timeout while connecting to %s", dataBaseName)
+			case <-endTimeoutDetection:
+				return
+			}
+			time.Sleep(10 * time.Millisecond)
+		}
 	}()
-	err = dbConnWriter.Ping()
+	err := dbConn.Ping()
 	if err != nil {
-		logger.Fatal(err)
```

Comment:
This is an intentional behavior change. A db that cannot be pinged most likely will not work so why should we not quit here?